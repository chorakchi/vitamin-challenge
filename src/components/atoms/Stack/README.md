The Stack component manages layout of immediate children along the vertical or horizontal axis with optional spacing and/or dividers between each child.

Stack is concerned with one-dimensional layouts, while flex-box that handles two-dimensional layouts. The default direction is row which stacks children horizontally.

To control space between children, use the spacing prop. The spacing value can be any number, including decimals and any string. The prop is converted into a CSS property 

## Direction
By default, Stack arranges items horizontally in a row. However, the direction prop can be used to position items vertically in a column as well.

## Dividers
Use the divider prop to insert an element between each child. This works particularly well with the Divider component.

## Interactive
Below is an interactive demo that lets you explore the visual results of the different settings: