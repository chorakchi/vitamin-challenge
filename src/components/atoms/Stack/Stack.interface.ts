import { HTMLAttributes } from "react";

export interface StackProps extends HTMLAttributes<HTMLElement> {
  direction?: "column" | "column-reverse" | "row" | "row-reverse";
  spacing?: number;
}
