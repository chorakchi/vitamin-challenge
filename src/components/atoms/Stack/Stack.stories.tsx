import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Stack } from "./Stack";
// @ts-ignore
import readme from "./README.md";

export default {
  title: "Atoms/Stack",
  component: Stack,
  parameters: {
    backgrounds: {
      default: "light",
    },
    docs: {
      description: {
        component: readme,
      },
    },
  },
} as ComponentMeta<typeof Stack>;

const Template: ComponentStory<typeof Stack> = (args) => <Stack {...args} />;

export const Stack_ = Template.bind({});
Stack_.args = {
  children: [
    <div className="stories-stack-block">Block 1</div>,
    <div className="stories-stack-block">Block 2</div>,
    <div className="stories-stack-block">Block 3</div>,
    <div className="stories-stack-block">Block 4</div>,
  ],
};
