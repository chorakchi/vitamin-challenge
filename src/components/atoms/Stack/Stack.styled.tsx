import styled from "styled-components";
import { StackProps } from "./Stack.interface";

export const StackStyled = styled.div<StackProps>`
  flex-direction: ${({ direction }) => direction || "row"};
  display: flex;
  gap: ${({ spacing }) => spacing + "px" || 0};
`;
