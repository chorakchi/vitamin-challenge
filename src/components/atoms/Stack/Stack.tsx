import { StackStyled } from "./Stack.styled";
import { StackProps } from "./Stack.interface";
export const Stack: React.FC<StackProps> = ({ ...props }: StackProps) => {
  return <StackStyled {...props} />;
};
