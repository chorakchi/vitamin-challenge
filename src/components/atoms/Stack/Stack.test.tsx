import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { Stack } from "./Stack";

describe("Stack", () => {
  it("should render the Stack", () => {
    const subject = shallow(<Stack>Stack</Stack>);
    expect(subject).toHaveLength(1);
  });

  it("should render correctly and match to the Snapshot", () => {
    const subject = renderer.create(<Stack>Stack</Stack>).toJSON();
    expect(subject).toMatchSnapshot();
  });
});
