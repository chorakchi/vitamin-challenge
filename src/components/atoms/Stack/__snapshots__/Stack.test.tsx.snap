// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack should render correctly and match to the Snapshot 1`] = `
<div
  className="Stackstyled__StackStyled-sc-1adxq1u-0 jWmaRX"
>
  Stack
</div>
`;
